models/readme.md

```diff
@@ -1 +1,164 @@
+
+### The model.py File
+
+Each model in the MLAgility benchmark is hosted in a `model.py` file. This file is a Python script that must meet the following requirements:
+1. Instantiate at least one model and invoke it against some inputs. The size and shape of the inputs will be used for benchmarking.
+1. Provide a docstring that provides information about where the model was sourced from.
+
+The file should be named according to the model within. For example, our file for the BERT model is named `bert.py`.
+
+Each `model.py` can optionally include a set of [labels](#labels) and [parameters](#parameters).
+
+### Labels
+
+MLAgility uses labels to help organize the results data. Labels must be in the first line of the Python file and start with `# labels: `
+
+Each label must have the format `key::value1,value2,...`
+
+Example:
+
+```
+# labels: author::google test_group::daily,monthly
+```
+     
+Labels are saved in your cache directory and can later be retrieved using the function `mlagility.common.labels.load_from_cache()`, which receives the `cache_dir` and `build_name` as inputs and returns the labels as a dictionary. 
+
+### Parameters
+
+MLAgility supports parameterizing models so that you can sweep over interesting properties such as batch size, sequence length, image size, etc. The `mlagility.parser.parse()` method parses a set of standardized parameters that we have defined and provides them to the model as it is instantiated.
+
+For example, this code would retrieve user-defined `batch_size` and `max_seq_length` (maximum sequence length) parameters.
+
+```
+from mlagility.parser import parse
+parse(["batch_size", "max_seq_length"])
+```
+
+You can pass parameters into a benchmarking run with the `--script-args` argument to `benchit`. For example, the command:
+
+```
+benchit bert.py --search-dir transformers_pytorch --script-args="--batch-size=8,--max-seq-length=128"
+```
+
+would set `batch_size=8` and `max_seq_length=128` for that benchmarking run.
+
+
+The standardized set of arguments is:
+
+- General args
+    - "batch_size": Arg("batch_size", default=1, type=int),
+    - "max_seq_length": Arg("max_seq_length", default=128, type=int),
+    - "max_audio_seq_length": Arg("max_audio_seq_length", default=25600, type=int),
+    - "height": Arg("height", default=224, type=int),
+    - "num_channels": Arg("num_channels", default=3, type=int),
+    - "width": Arg("width", default=224, type=int),
+- Args for Graph Neural Networks
+    - "k": Arg("k", default=8, type=int),
+    - "alpha": Arg("alpha", default=2.2, type=float),
+    - "out_channels": Arg("out_channels", default=16, type=int),
+    - "num_layers": Arg("num_layers", default=8, type=int),
+    - "in_channels": Arg("in_channels", default=1433, type=int),- 
```

Comment:
We might want to review those eventually, as a few of those GNN parameters are only used in 1-2 models and are not very descriptive.

Reply:
Yeah they should probably have comments or something

Reply:
I added definitions for the general args. Made an issue for @ramkrishna2910 to add definitions to the GNN args https://github.com/groq/mlagility/issues/78